substrate/client/network/sync/src/strategy/state_sync.rs

```diff
@@ -89,22 +89,62 @@ pub enum ImportResult<B: BlockT> {
 	BadResponse,
 }
 
-/// State sync state machine. Accumulates partial state data until it
-/// is ready to be imported.
-pub struct StateSync<B: BlockT, Client> {
-	target_block: B::Hash,
+struct StateSyncMetadata<B: BlockT> {
```

Comment:
minor nit; Would be useful to have some documatation on this struct and exposed methods